cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/LineageOperationsProcessor.java

```diff
@@ -211,10 +219,12 @@ private void addMergeOperation(Set<String> stageInputs, Map<String, Operation> p
         Iterator<String> stageFieldPairIter = Splitter.on(".").omitEmptyStrings().trimResults().split(field).iterator();
         String stageName = stageFieldPairIter.next();
         if (stageFieldPairIter.hasNext() && stageOperations.keySet().contains(stageName)) {
-          field = stageFieldPairIter.next();
+          actualField = stageFieldPairIter.next();
           fieldsForJoin.add(field);
```

Comment:
this happens in both if and else statements, should we move it out and remove the else?

Reply:
actually, this doesn't even look like it is used?